cocos2d/Platforms/CCAppController.h

```diff
@@ -0,0 +1,107 @@
+//
+//  CCAppController.h
+//  cocos2d
+//
+//  Created by Donald Hutchison on 13/01/15.
+//
+//
+
+#import <Foundation/Foundation.h>
+#import "CCDirector.h"
+
+@class NSWindow;
+@class CCGLView;
+@class CCScene;
+
+/**
+ *  
+ CCAppController serves as the setup point for the application and prepares cocos2d to run with various platform specific settings.
+ It exists outside of cocos scene management, and as such can be used to start application specific logic.
+ 
+ An instance of this class (or a subclass) should be invoked from the entry point of execution for each platform. These points are:
+ 
+ * AppDelegate#application:didFinishLaunchingWithOptions: (iOS)
+ * AppDelegate#applicationDidFinishLaunching: (Mac)
+ * CCActivity#run (Android)
+  
+ For SpriteBuilder projects the above files will be generated automatically with the correct invocations. 
+ 
+ Caveats exist for Android, as unlike with other platforms the GL context is not initialized before the setup process begins. Care must 
+ be taken not to perform setup logic that requires access to GL functions before initialization is complete.
+
+ */
+
+@interface CCAppController : NSObject
+
+/**
+*  The view in which the cocos nodes and scene graph are rendered
+*/
+@property (weak) CC_VIEW<CCView> *glView;
+
+/// -----------------------------------------------------------------------
+/// @name Mac Specific
+/// -----------------------------------------------------------------------
+
+/**
+*  The application window (set from .xib)
+*/
+@property (weak) NSWindow *window;
```

Comment:
Not quite sure why this is here and only for Mac. If you really need a reference to the window that holds a view, you can use `NSView.window`.